### PR TITLE
Feature flag to support the legacy column style -- bug fixes

### PR DESCRIPTION
**Pull request by istreeter**

### CODE REVIEW

File: modules/core/src/main/scala/com.snowplowanalytics.snowplow.bigquery/processing/LegacyColumns.scala [hunk 1]
```diff
@@ -160,6 +160,24 @@ object LegacyColumns {
         BadRow.LoaderIgluError(processor, BadRowFailure.LoaderIgluErrors(nel), BadPayload.LoaderPayload(event))
       }.toEither
 
+  /**
+   * Replaces any Json type with a String type
+   *
+   * This must be run after transformation, but before handling schema evolution. It is needed
+   * because the legacy loader used String columns where the v2 loader would create Json columns
+   */
+  def dropJsonTypes(v2Field: V2Field): V2Field = {
```
Comment (pondzix) [inline modules/core/src/main/scala/com.snowplowanalytics.snowplow.bigquery/processing/LegacyColumns.scala:169]: I'm wondering, what happens if in `legacyTypeToV2Type` [here](https://github.com/snowplow-incubator/snowplow-bigquery-loader/blob/819f267ed3795adef0232e5de9e7727a8aaa8b54/modules/core/src/main/scala/com.snowplowanalytics.snowplow.bigquery/processing/LegacyColumns.scala#L184) if we map to `V2Type.String` instead? So don't use `V2Type.Json` anywhere in legacy columns. What is then broken?

If we can  use `V2Type.String`, then do we still need to handle JSONs explicitly later (like here)?

File: modules/core/src/main/scala/com.snowplowanalytics.snowplow.bigquery/processing/LegacyColumns.scala [hunk 2]
```diff
@@ -125,8 +125,8 @@ object LegacyColumns {
               case TabledEntity.Context       => LegacyMode.Repeated
             }
             val columnName  = legacyColumnName(entityType, schemaKey)
-            val legacyField = LegacyField.build(columnName, schema, false).setMode(mode).normalized
-            val v2Field     = legacyFieldToV2Field(legacyField).copy(accessors = Set.empty)
+            val legacyField = LegacyField.build(columnName, schema, false).setMode(mode)
+            val v2Field     = V2Field.normalize(legacyFieldToV2Field(legacyField)).copy(accessors = Set.empty)
```
Comment (pondzix) [inline modules/core/src/main/scala/com.snowplowanalytics.snowplow.bigquery/processing/LegacyColumns.scala:129]: What is the reason of this change?

Now we normalize legacy fields using V2 normalization. Just want to be sure it's 100% safe 😁

File: modules/core/src/main/scala/com.snowplowanalytics.snowplow.bigquery/processing/LegacyColumns.scala (hunk 1 above)
Comment (istreeter) [reply modules/core/src/main/scala/com.snowplowanalytics.snowplow.bigquery/processing/LegacyColumns.scala:169]: It's because the legacy ddl contains a String type under several circumstances:
1. The field is a really a string field
2. An object with no properties defined
3. An enum comprising numbers AND strings

Let's consider example 2.  If the json schema contains:

```
"a": {"type": "object"}
```

...and the event contains:

```
"a": {"xxx": "yyy"}
```

...and if we returned `V2Type.String` [on the line you said](https://github.com/snowplow-incubator/snowplow-bigquery-loader/blob/819f267ed3795adef0232e5de9e7727a8aaa8b54/modules/core/src/main/scala/com.snowplowanalytics.snowplow.bigquery/processing/LegacyColumns.scala#L184), then the casting would fail [on this line](https://github.com/snowplow/schema-ddl/blob/414a2d76902b8653433154a1c7029b15768ed888/modules/core/src/main/scala/com.snowplowanalytics/iglu.schemaddl/parquet/Caster.scala#L67) so we'd get a failed event.

So I work around that problem by returning `V2Type.Json` and then handling the consequences later.  It's hacky, but it works.

File: modules/core/src/main/scala/com.snowplowanalytics.snowplow.bigquery/processing/LegacyColumns.scala (hunk 2 above)
Comment (istreeter) [reply modules/core/src/main/scala/com.snowplowanalytics.snowplow.bigquery/processing/LegacyColumns.scala:129]: If we normalize the v1 field, then we drop information about whether the original field name was snake case or camel case.  So if the original field name was camel case then the casting fails [on this line](https://github.com/snowplow/schema-ddl/blob/414a2d76902b8653433154a1c7029b15768ed888/modules/core/src/main/scala/com.snowplowanalytics/iglu.schemaddl/parquet/Caster.scala#L176).

I discovered this when I tried to load some events with the yauaa context.

Hmmm, maybe I should change [these test schemas](https://github.com/snowplow-incubator/snowplow-bigquery-loader/tree/v2/modules/core/src/test/resources/iglu-client-embedded/schemas/myvendor/myschema/jsonschema) to include some camel case fields.  And then this PR would be fixing a failing a test.

Comment (pondzix) [reply modules/core/src/main/scala/com.snowplowanalytics.snowplow.bigquery/processing/LegacyColumns.scala:129]: Ok, that makes sense!

Last question - why do we need `.copy(accessors = Set.empty)` here?

Comment (istreeter) [reply modules/core/src/main/scala/com.snowplowanalytics.snowplow.bigquery/processing/LegacyColumns.scala:129]: It doesn't make any difference to how the loader works.  But it makes it easier to write tests.

If I remove the `.copy(...)`, then I would also have to change [this line in the Spec](https://github.com/snowplow-incubator/snowplow-bigquery-loader/blob/819f267ed3795adef0232e5de9e7727a8aaa8b54/modules/core/src/test/scala/com.snowplowanalytics.snowplow.bigquery/processing/LegacyColumnsResolveSpec.scala#L65) to be like...

```scala
val expectedField = Field("unstruct_event_myvendor_myschema_7_0_0", expectedStruct, Nullable, "unstruct_event_myvendor_myschema_7_0_0")
```

(+ lots of other places)

...which is fine, but it's a bit annoying to repeat the field name each time in the expected field.

Comment (istreeter) [reply modules/core/src/main/scala/com.snowplowanalytics.snowplow.bigquery/processing/LegacyColumns.scala:129]: Actually ignore that... it's completely wrong.  I'll try removing the `.copy(....)`, because it might work.
